README.md

```diff
@@ -1,2 +1,96 @@
+[![Godoc](https://godoc.org/github.com/vapor-ware/synse-client-go/synse?status.svg)](https://godoc.org/github.com/vapor-ware/synse-client-go/synse)
+[![Go Report Card](https://goreportcard.com/badge/github.com/vapor-ware/synse-client-go)](https://goreportcard.com/report/github.com/vapor-ware/synse-client-go)
+
 # synse-client-go
-Golang client for the Synse Server HTTP API
+
+This repo contains a HTTP and WebSocket client for interacting with Synse
+Server API, written in Go. For more information about the API, please visit its
+specification at [HTTP
+API](https://github.com/vapor-ware/synse-server/blob/master/proposals/v3/api.md),
+and [WebSocket
+API](https://github.com/vapor-ware/synse-server/blob/master/proposals/v3/api-websocket.md).
+
+## Installing
+
+In order to install this package, you can clone the repo, `cd` into the repo
+root, and set it up via `make`:
+```
+$ git clone https://github.com/vapor-ware/synse-client-go.git
+$ cd synse-client-go
+$ make setup
+```
+
+## Using
+
+### Initializing
+
+For both HTTP client and WebSocket client, in order to initialize their
+instances, we need to pass in the [configuration
+options](https://godoc.org/github.com/vapor-ware/synse-client-go/synse#Options),
+namely the Synse Server's address that we need to interface wish, other
+associated HTTP or WebSocket configs, and TLS communication. For example:
+```go
+import "github.com/vapor-ware/synse-client-go/synse"
+
+func main() {
+  opts := &synse.Options{
+    Address: "localhost",
+  }
+
+  client, err := NewHTTPClientV3(opts)
+  // or client, err := NewWebSocketClientV3(opts)
+}
+```
+
+### API
+
+Below is the map of client methods with their corresponding HTTP and WebSocket API.
+
+| Method | HTTP endpoint | WebSocket request |
+| ------ | ------------- | ----------------- |
+| `Status()` | `/test` | `request/status` |
+| `Version()` | `/version` | `request/version` | `Version()` |
+| `Config()` | `/v3/config` | `request/config` | `Config()` |
```

Comment:
I don't think the 4th column here will get rendered since there are only 3 header columns defined